src/server/game/Combat/ThreatManager.h

```diff
@@ -15,284 +14,154 @@
+
+        static float CalculateModifiedThreat(float threat, Unit const* victim, SpellInfo const* spell);
+
+        ThreatManager(Unit* owner);
+        void Update(uint32 tdiff); // called from Creature::Update (only creatures can have their own threat list)
+
+        Unit* GetOwner() const { return _owner; }
+        Unit* GetCurrentVictim() const { return _currentVictimRef->GetVictim(); }
+        Unit* GetAnyTarget() const;
+        Unit* SelectVictim();
+
+        bool IsEngaged() const { return _ownerEngaged; }
+        bool IsThreatListEmpty(bool includeOffline = false) const;
+        bool IsThreatenedBy(Unit const* who, bool includeOffline = false) const;
+        float GetThreat(Unit const* who, bool includeOffline = false) const;
+        std::vector<ThreatReference*> const& GetThreatList() const { return _sortedThreatList; } // NB: Not necessarily structured in any way unless you call HeapifyThreatList()/SortHeapifiedThreatList() first!
+
+        bool IsThreateningAnyone(bool includeOffline = false) const;
+        std::unordered_map<ObjectGuid, ThreatReference*> const& GetThreatenedByMeList() const { return _threatenedByMe; }
+
+        void UpdateOnlineStates(bool meThreateningOthers = true, bool othersThreateningMe = true);
+        ///== AFFECT MY THREAT LIST ==
+        void HeapifyThreatList();
+        void SortHeapifiedThreatList();
+        void AddThreat(Unit* target, float amount, SpellInfo const* spell = nullptr, bool ignoreModifiers = false, bool ignoreRedirects = false);
+        void ScaleThreat(Unit* target, float factor);
+        void ModifyThreatByPercent(Unit* target, int32 percent) { ScaleThreat(target, 0.01f*float(100+percent)); }
+        void ResetThreat(Unit* target) { ScaleThreat(target, 0.0f); }
+        void MatchHighestThreat(Unit* target);
+        void TauntUpdate();
+        void ResetAllThreat(); // Keeps threat list entries, just sets them to have no threat
+        void ClearThreat(Unit* target);
+        void ClearAllThreat(); // Removes the actual threat list entries
+
+        void SendThreatListToClients() const;
+
+        ///== AFFECT OTHERS' THREAT LISTS ==
+        void ScaleMyThreat(float factor);
+        void RemoveMeFromThreatLists();
+        void UpdateMyTempModifiers(); // re-calculates the temporary threat modifier from auras on myself
+        void UpdateMySpellSchoolModifiers();
+        void RegisterRedirectThreat(uint32 spellId, ObjectGuid const& victim, uint32 pct);
+        void UnregisterRedirectThreat(uint32 spellId);
+        void UnregisterRedirectThreat(uint32 spellId, ObjectGuid const& victim);
+        void ForwardThreatForAssistingMe(Unit* assistant, float baseAmount, SpellInfo const* spell = nullptr, bool ignoreModifiers = false);
     private:
-        void _addThreat(Unit* victim, float threat);
+        Unit* const _owner;
+        bool _ownerEngaged;
+
+        // send opcodes (all for my own threat list)
+        void sendClearAllThreatToClients() const;
```

Comment:
Do we have any way to distinguish between private/public member functions in CS?
In this PR, lowercase = private, uppercase = public, though I can obviously refactor that if required (just hearing conflicting versions of "CS" here).